### PR TITLE
[TR-5020] Long environment variable name can be created, but can't be…

### PR DESCRIPTION
**Pull request by tanmibts**
Added documentation for restrictions on environment variable names, as well as updated that values can be numbers, strings, or booleans.

### CODE REVIEW

File: building/environment-variables.md [hunk 1]
```diff
@@ -11,7 +11,14 @@ Environment variables are stored encrypted-at-rest.
 
 You can define an environment variable schema at **Settings &gt; App Info &gt; Environment variables schema**.
 
-Keys and values can be any valid string.
+Environment variable names (keys) must be strings, with the following restrictions:
+- It can't be blank.
+- Its length can't exceed 30 characters.
+- It must be unique.
+- All characters must follow `[a-zA-Z0-9._-]`.
```
Comment (mooreds) [inline building/environment-variables.md]: Do you mean that only those characters are allowed?

File: building/environment-variables.md [hunk 2]
```diff
@@ -11,7 +11,14 @@ Environment variables are stored encrypted-at-rest.
 
 You can define an environment variable schema at **Settings &gt; App Info &gt; Environment variables schema**.
 
-Keys and values can be any valid string.
+Environment variable names (keys) must be strings, with the following restrictions:
+- It can't be blank.
+- Its length can't exceed 30 characters.
+- It must be unique.
```
Comment (mooreds) [inline building/environment-variables.md]: unique across the application? across the account? across all transposit applications?

File: building/environment-variables.md [hunk 3]
```diff
@@ -11,7 +11,14 @@ Environment variables are stored encrypted-at-rest.
 
 You can define an environment variable schema at **Settings &gt; App Info &gt; Environment variables schema**.
 
-Keys and values can be any valid string.
+Environment variable names (keys) must be strings, with the following restrictions:
```
Comment (mooreds) [inline building/environment-variables.md]: are they case sensitive? I think so, but would be worth calling out.

File: building/environment-variables.md [hunk 4]
```diff
@@ -11,7 +11,14 @@ Environment variables are stored encrypted-at-rest.
...
+- It can't be blank.
+- Its length can't exceed 30 characters.
+- It must be unique.
+- All characters must follow `[a-zA-Z0-9._-]`.
+- The first character must follow `[a-zA-Z._]`.
```
Comment (mooreds) [inline building/environment-variables.md]: same q as above.

File: building/environment-variables.md [hunk 5]
```diff
@@ -11,7 +11,14 @@ Environment variables are stored encrypted-at-rest.
...
+
+Environment variable values can be any valid string, number, or boolean.
```
Comment (mooreds) [inline building/environment-variables.md]: There are no length restrictions on variable values? Are they allowed to be null? Any character restrictions?

File: building/environment-variables.md (hunk 1 above)
Comment (tanmibts) [reply building/environment-variables.md]: Yes! Please do suggest a way to make it more clear -- I also think it's a lil clunky.

File: building/environment-variables.md (hunk 5 above)
Comment (tanmibts) [reply building/environment-variables.md]: @blazarus can you answer these?

File: building/environment-variables.md (hunk 1 above)
Comment (mooreds) [reply building/environment-variables.md]: How about "Only alphanumeric characters and the period, underscore and dash are allowed."

File: building/environment-variables.md (hunk 5 above)
Comment (blazarus) [reply building/environment-variables.md]: oo sorry I missed this! Looking at the database schema, the `value` column for user_settings `varchar(255)` characters while the `value` column in the table for env is just `text`, which looks like it has a max size of 64kb in mysql.
I feel like the intention was to move user_settings to the same schema as env, but that never happened. So I'm not sure what makes the most sense for documentation - I'd probably lean towards leaving it undocumented for now, but don't feel strongly.

I believe environment variables are null if they are unset and don't have a default value. No character restrictions as far as I know. @tuan-nguyen-tps or @m-raymond-williams may have more insight here though.

**Pull request merged**